packages/pinia/src/rootStore.ts

```
@@ -43,7 +43,7 @@ interface _SetActivePinia {
  * Get the currently active pinia if there is any.
  */
 export const getActivePinia = () =>
-  (hasInjectionContext() && inject(piniaSymbol)) || activePinia
+  (hasInjectionContext() && inject(piniaSymbol)) || (import.meta.server ? throw new Error("Cannot get active pinia as it does not find context") : activePinia)
```

Thanks for the PR. This could be changed into a non breaking change by just showing an error and still returning the active pinia. The error should be in development only (like other warnings) and should not rely on `import` as it might not work in all scenarios

---

I think failing fast is the better approach when the global `activePinia` is accessed on the server, because:
1. failing immediately makes the problem obvious rather than silently continuing with potentially incorrect behavior.
2. allowing it to continue by returning `activePinia` could lead to dangerous state sharing between requests, which is particularly problematic in server environments.

maybe making the message is more clear would help huge

---

That would be a breaking change. Also, I do prefer the error message because it feels less frustrating to users

---

I will leave the decision to you,

However, please keep in mind that the developers tend to ignore error messages, so they will make the same mistakes again and again, I would prefer it to fail that will indicate the mistake much faster.